modules/mobile/app/controllers/mobile/v0/pre_need_burial_controller.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'chip/service'
+require 'preneeds/service'
+
+module Mobile
+  module V0
+    class PreNeedBurialController < ApplicationController
+      FORM = '40-10007'
+
+      def create
+        validate!
+
+        @form = ::Preneeds::BurialForm.new(burial_form_params)
+        @resource = client.receive_pre_need_application(@form)
+
+        create_local_preneed_submission(@resource)
+
+        render json: Mobile::V0::PreNeedBurialSerializer.new(@resource)
+      end
+
+      private
+
+      def create_local_preneed_submission(resource)
+        ::Preneeds::PreneedSubmission.create!(
+          tracking_number: resource.tracking_number,
+          application_uuid: resource.application_uuid,
+          return_description: resource.return_description,
+          return_code: resource.return_code
+        )
+      end
+
+      def burial_form_params
+        params.require(:application).permit(
+          :application_status, :has_currently_buried, :sending_code,
+          preneed_attachments: ::Preneeds::PreneedAttachmentHash.permitted_params,
+          applicant: ::Preneeds::Applicant.permitted_params,
+          claimant: ::Preneeds::Claimant.permitted_params,
+          currently_buried_persons: ::Preneeds::CurrentlyBuriedPerson.permitted_params,
+          veteran: ::Preneeds::Veteran.permitted_params
+        )
+      end
+
+      def validate!
+        json_params = JSON.parse(params.to_json)
```

Comment:
Doesn't this just convert it from a hash to json then back to a hash? I'm not sure why we need to do this.

Reply:
I think I see the issue. The reason you have to do this json conversion is because when you do 
`params.require(:application).permit`
it doesn't actually permit application, only the specific portions you permitted. As a result, you can't use `burial_form_params` here because it's missing `application` but if you try to do params.to_h (which is done implicitly when you dig into params), it complains that there are unpermitted properties in the hash. It looks like you need the permitted params to be structured without application in order to create the burial form. 

TBH, I don't totally understand why we have to create a burial form for validation then another for completing the process. I assume that's logic you copied from elsewhere. Ideally, I think it should all be done in one object. I'm sure there are multiple ways of fixing this, but assuming that we can't use the same strong params for object creation and validation, I'd suggest avoiding the json conversion and doing this instead:

```
        normalized_params = params.to_unsafe_hash
        normalized_params.dig('application', 'veteran', 'race').transform_values! do |v|
          ActiveRecord::Type::Boolean.new.cast(v)
        end
```

One reason I say that is that the json conversion is confusing. Another reason is that the variable name `json_params` is misleading because it's not json. It might also be worth a comment explaining why the different params are necessary and/or why validation and object creation can't be done on the same burial needs form object.

Reply:
It was logic copied from the web controller. I assumed there I was able to get it to use the form instance with a simple `to_json` so that cleaned it up a lot.